app/webhook.go

```diff
@@ -145,15 +150,26 @@ func (a *App) TriggerWebhook(payload *model.OutgoingWebhookPayload, hook *model.
 	}
 }
 
-func (a *App) doOutgoingWebhookRequest(url string, body io.Reader, contentType string) (*model.OutgoingWebhookResponse, error) {
-	req, err := http.NewRequest("POST", url, body)
+func (a *App) doOutgoingWebhookRequest(url string, body io.Reader, contentType, secretToken string) (*model.OutgoingWebhookResponse, error) {
+	bodyBytes, err := ioutil.ReadAll(body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read outgoing webhook request payload: %w", err)
+	}
+	bodyReader := bytes.NewReader(bodyBytes)
+
+	req, err := http.NewRequest("POST", url, bodyReader)
 	if err != nil {
 		return nil, err
 	}
 
 	req.Header.Set("Content-Type", contentType)
 	req.Header.Set("Accept", "application/json")
 
+	now := strconv.FormatInt(time.Now().Unix(), 10)
+	hmacDigest := model.GenerateHmacSignature(&bodyBytes, now, secretToken)
+	req.Header.Set("Timestamp", now)
+	req.Header.Set("Signature", hmacDigest)
```

Comment:
@jupenur, do you have thoughts on the best headers to use here? I know Slack [uses](https://api.slack.com/docs/verifying-requests-from-slack) `X-Slack-Signature` and `X-Slack-Request-Timestamp`, so there's obviously no standard here: but wondering we should namespace similarly? Or is there another standard we might consider adopting for nomenclature here?

Reply:
Based on [RFC 6648](https://tools.ietf.org/html/rfc6648), I'd go for something like `Mattermost-Signature` and `Mattermost-Timestamp`. I don't think there are any established conventions for webhooks specifically, though.

Reply:
I have changed to what @jupenur  suggested.